1st draft GitHub plugin page

s-ben opened the pull request:
Created a rough first draft of the GitHub plugin page. Mostly porting content from this [PR](https://github.com/sourcecred/sourcecred/pull/1720), as I think it does a good job explaining how Cred flows, defining the nodes, caveats, etc. In fact, I think we could use the basic structure in this page for the other plugins, unless anyone has objections/better ideas. 

Changes made:
- Light rewriting throughout, e.g. capitalizing 'Cred' and replacing 'pulls' with 'PRs', per our decision in another doc
- Removed mirror module section, as it seemed more dev focused
- Added 'Cred flow' section with placeholder for example of diagram of a typical GitHub flow (open to ditching)

Questions:
- Is the plugin itself still in beta?
- Have we added any features since this was last updated (last commit was April 14th)? Any that we expect to be in before beta launch? Minting Cred on PRs??
- Do we want to include configuration info here, or in a separate page?

decentralion commented on docs/setup/plugins/github.md:
```diff
@@ -2,4 +2,223 @@
+servers. That means that if content is deleted on GitHub, it will also
+disappear from the GitHub plugin (after a cache refresh).
+
+## Status and Caveats
```
Let's move the caveats to the bottom.

s-ben replied:
Done

decentralion commented on docs/setup/plugins/github.md:
```diff
@@ -2,4 +2,223 @@
+
+## Status and Caveats
+
+The GitHub plugin is still in beta. It tends to assign reasonable Cred scores
+to contributors. However, it has difficulty assigning Cred scores to Pull
+Requests (PRs). This is because, in general, the plugin mints Cred based on raw
+activity. Depending on weight choices, every pull will mint a fixed amount of
+Cred, likewise for every comment, issue, review, and so forth.
+
+This is problematic because, to put it simply, not all activity is equally
+valuable. Minting Cred directly on raw activity encourages contributors to
+focus on quantity over quality. As an example, a poorly thought out PR
+ (which requires lots of review and feedback to fix) will generate more
+Cred than a clean, elegant pull request (which merges with a minimum of fuss).
```
This is no longer really true. By default, it mints a fixed amount of Cred for every **merged** PR. It also mints a smaller amount of Cred for each review.

s-ben replied:
Reworded to reflect current state. Could use a review

decentralion commented on docs/setup/plugins/github.md:
```diff
@@ -2,4 +2,223 @@
+ (which requires lots of review and feedback to fix) will generate more
+Cred than a clean, elegant pull request (which merges with a minimum of fuss).
+
+We intend to improve the robustness of the GitHub plugin by
+adding better heuristics for assigning Cred, e.g.:
+- Minting Cred when PRs are merged, rather than when they are created.
+- Allowing custom labels that influence the Cred minted to PRs.
+- Modifying Cred minting based on metrics like the size of the PR.
```
We  can take away the first bullet, since that's been fixed. Let's keep the next two.

s-ben replied:
Removed first bullet from both lists, as both addressed minting on merge

decentralion commented on docs/setup/plugins/github.md:
```diff
@@ -2,4 +2,223 @@
+
+[example image]
+
+Here we can see Bob reporting a bug by filing an Issue. Alice then 
+submits a PR fixing the Issue. Pam reviews the fix and submits a 
+PR review....SB: how complicated do we want the example to be? A simple
+flow like above seems fine, but wondering if a more complex flow could
+be illustrative too?  
```
I think a simple flow is fine.

s-ben replied:
Want to see if ben.oxmo wants to take a stab at a diagram before I do it. If we're waiting on that, maybe I can just remove the example so we can merge? Then do example as separate PR?

decentralion commented on docs/setup/plugins/github.md:
```diff
@@ -2,4 +2,223 @@
+
+- **Repo**:
+
+A GitHub repository, e.g. [sourceCred/sourceCred]. The repo node will be
```
`sourcecred/sourcecred`. we don't capitalize the S or the C in the repository name.

(here and throughout the rest of the doc)

s-ben replied:
Find and Replace fail. Got a little cocky there with the 'Replace All' button, flying too close to the sun...

s-ben commented:
@decentralion, addressed comments. I think we're converging. 

>Since I wrote the doc this is ported from, maybe add a paired-with edge to me in the PR description?

Is this accidental double dipping with your original sourcecred/sourcecred [PR](https://github.com/sourcecred/sourcecred/pull/1720)? Makes sense to split the Cred and fine with doing paired-with edge, but wondering the effect of Cred split across repos.

s-ben commented:
@decentralion, sounds good to me. Added TO DO in place of example and put a link in the description to the PR for the original GitHub README.
